src/navigation/navigation.cc

```diff
@@ -818,17 +820,26 @@ bool Navigation::Run(const double& time,
   }
 
   // Get Carrot and check if done
-  const Vector2f carrot = GetCarrot();
+  try {
```

Comment:
We don't use exceptions in our code. If something is throwing exceptions in our code, we need to fix it. If this catches an exception from a third party source, please add a comment to explain what it is.